warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -4,89 +4,97 @@
+import datawave.query.util.sortedset.RewritableSortedSetImpl;
 
 /**
  * This iterator will filter documents based on uniqueness across a set of configured fields. Only the first instance of an event with a unique set of those
- * fields will be returned. This transform is thread safe.
+ * fields will be returned unless mostRecentUnique is specified in which case the most recent instance of an event will be returned. This transform is thread
+ * safe.
  */
 public class UniqueTransform extends DocumentTransform.DefaultDocumentTransform {
 
     private static final Logger log = Logger.getLogger(UniqueTransform.class);
 
     private BloomFilter<byte[]> bloom;
-    private UniqueFields uniqueFields;
+    private UniqueFields uniqueFields = new UniqueFields();
     private Multimap<String,String> modelMapping;
+    private HdfsBackedSortedSet<Entry<byte[],Document>> set;
+    private Iterator<Entry<byte[],Document>> setIterator;
 
-    public UniqueTransform(UniqueFields uniqueFields) {
-        this.uniqueFields = uniqueFields;
-        this.uniqueFields.deconstructIdentifierFields();
+    private UniqueTransform(UniqueFields uniqueFields) {
+        this.uniqueFields = uniqueFields.clone();
         this.bloom = BloomFilter.create(new ByteFunnel(), 500000, 1e-15);
         if (log.isTraceEnabled()) {
             log.trace("unique fields: " + this.uniqueFields.getFields());
         }
     }
 
     /**
-     * Create a new {@link UniqueTransform} that will capture the reverse field mapping defined within the model being used by the logic (if present).
+     * Update the configuration of this transform. If the configuration is actually changing, then the bloom filter will be reset as well.
      *
-     * @param logic
-     *            the logic
      * @param uniqueFields
-     *            the set of fields to find unique values for
+     *            The new set of unique fields.
+     * @param model
+     *            The query model
      */
-    public UniqueTransform(BaseQueryLogic<Entry<Key,Value>> logic, UniqueFields uniqueFields) {
-        this(uniqueFields);
-        QueryModel model = ((ShardQueryLogic) logic).getQueryModel();
-        if (model != null) {
-            modelMapping = HashMultimap.create();
-            // reverse the reverse query mapping which will give us a mapping from the final field name to the original field name(s)
-            for (Map.Entry<String,String> entry : model.getReverseQueryMapping().entrySet()) {
-                modelMapping.put(entry.getValue(), entry.getKey());
+    public void updateConfig(UniqueFields uniqueFields, QueryModel model) {
+        // only reset the bloom filter if changing the field set
+        if (!this.uniqueFields.equals(uniqueFields)) {
+            this.uniqueFields = uniqueFields.clone();
+            log.info("Resetting unique fields on the unique transform");
+            this.bloom = BloomFilter.create(new ByteFunnel(), 500000, 1e-15);
```

Comment:
Perhaps these arguments would be worth making configurable, maybe in a future iteration.